sys/tlk_device.txt

```diff
@@ -0,0 +1,63 @@
+# Copyright 2016 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+# Open Trusted Execution driver for /dev/tlk_device
+# Reference source code:
+# https://android.googlesource.com/kernel/tegra/+/android-tegra-dragon-3.18-marshmallow-dr-dragon/security/tlk_driver/ote_protocol.h
+
+include <linux/ioctl.h>
+include <linux/types.h>
+include <security/tlk_driver/ote_protocol.h>
+
+syz_open_dev$tlk_device(dev strconst["/dev/tlk_device"], id const[0], flags flags[open_flags]) fd[tlk_device]
+
+ioctl$TE_IOCTL_OPEN_CLIENT_SESSION(fd fd[tlk_device], cmd const[TE_IOCTL_OPEN_CLIENT_SESSION], arg ptr[inout, te_cmd])
```

Comment:
Pass arg ptr[inout, te_opensession] directly. No need for the union.
The same for other ioctls.